### PR TITLE
Chore/allow options for fulfillment create

### PR DESCRIPTION
**Pull request by chrynuik**
When creating a Fulfillment in Shopify, requests are allowed to take up to 40s before timing out. Added options to `fulfillment.create/4` to override the `recv_timeout` option.

### CODE REVIEW

File: lib/shopify_api/rest.ex
Comment (stewart) [inline lib/shopify_api/rest.ex:60]: Could we also add this optional parameter to `put/3`, please?

File: CHANGELOG.md
Comment (stewart) [inline CHANGELOG.md:3]: Would be useful to note we've added the ability to pass a list of `HTTPoison` options as `options` to `Rest.post` and `Rest.put`.

**Review by stewart — COMMENTED** [review]
:+1: Do you think it makes sense to also rename the `shopify_api` config key from `http_timeout` to `recv_timeout` if that's all it is used for?

File: config/config.exs
Comment (stewart) [inline config/config.exs:34]: Should highlight this as a "BREAKING" change in the `CHANGELOG`.

File: CHANGELOG.md
Comment (stewart) [inline CHANGELOG.md:3]: Can you please add a secondary comment specifying that this is also available via the new optional 4th param to `ShopifyAPI.REST.Fulfillment.create`.

File: lib/shopify_api/rest/request.ex
Comment (hez) [inline lib/shopify_api/rest/request.ex:17]: This should be a compile env, since it is a module attribute

File: CHANGELOG.md
Comment (hez) [inline CHANGELOG.md:4]: 🍎 `recv_timeout` seems like a very HTTPoison specific name, are we sure we want to rename to that? I can't think of a better name off the top of my head right now.

Comment (hez) [reply CHANGELOG.md:4]: Thinking of this ticket https://github.com/pixelunion/elixir-shopifyapi/issues/308 and possibly moving to a different HTTP client library.

Comment (chrynuik) [reply CHANGELOG.md:4]: The reason why we picked this name is that there are two variables for HTTPoison `http_timeout` is to specify the time it takes for the connection to timeout before making the connection and `recv_timeout` is the time it takes to wait for the response.  [Docs](https://hexdocs.pm/httpoison/HTTPoison.Request.html)

Comment (hez) [reply CHANGELOG.md:4]: Good catch on the misnaming, what about something like `rest_receive_timeout`?

File: lib/shopify_api/rest/request.ex
Comment (hez) [reply lib/shopify_api/rest/request.ex]: Quick on the fix! <3

File: CHANGELOG.md
Comment (chrynuik) [reply CHANGELOG.md]: This makes sense. I have changed the name of the variable.

**Review by freemer — APPROVED** [review]
👍

**Review by stewart — APPROVED** [review]

**Comment by stewart** [comment]
Sorry, looks like I introduced a CHANGELOG conflict by merging #356 that needs to be resolved on this PR via rebase. Once that's done, I believe this is good to merge!

**Review by hez — APPROVED** [review]

**Pull request merged**
